packages/launchpad-auth/src/index.ts

```diff
@@ -6,57 +6,70 @@
  * @module launchpad-auth
  */
 
-const pick = require('lodash/pick');
-const {
+import pick from 'lodash/pick';
+import {
   getS3Object,
   s3Join,
   s3ObjectExists,
   s3PutObject
-} = require('@cumulus/aws-client/S3');
-const Logger = require('@cumulus/logger');
+} from '@cumulus/aws-client/S3';
+import Logger from '@cumulus/logger';
 
-const LaunchpadToken = require('./LaunchpadToken');
+import {
+  LaunchpadTokenParams,
+  TokenObject,
+  ValidateTokenResult
+} from './types';
+
+import LaunchpadToken from './LaunchpadToken';
+import { getEnvVar } from './utils';
 
 const log = new Logger({ sender: '@cumulus/launchpad-auth' });
 
 /**
  * Get S3 location of the Launchpad token
  *
- * @returns {Promise.<Object.<string, string>>} - S3 Bucket and Key where Launchpad token is stored
+ * @returns {Promise<Object<string, string>>} - S3 Bucket and Key where Launchpad token is stored
  *
  * @private
  */
-function launchpadTokenBucketKey() {
-  if (!(process.env.stackName && process.env.system_bucket)) {
-    throw new Error('must set environment variables process.env.stackName and process.env.system_bucket');
-  }
-
+function launchpadTokenBucketKey(): {
+  Bucket: string,
+  Key: string
+} {
+  const bucket = getEnvVar('system_bucket');
+  const stackName = getEnvVar('stackName');
   return {
-    Bucket: process.env.system_bucket,
-    Key: s3Join(process.env.stackName, 'launchpad/token.json')
+    Bucket: bucket,
+    Key: s3Join(stackName, 'launchpad/token.json')
   };
 }
 
 /**
  * Retrieve Launchpad token from S3
  *
- * @returns {Promise.<string>} - the Launchpad token, null if token doesn't exist or invalid
+ * @returns {Promise<string|undefined>}
+ *   the Launchpad token, undefined if token doesn't exist or invalid
  *
  * @async
  * @private
  */
-async function getValidLaunchpadTokenFromS3() {
+async function getValidLaunchpadTokenFromS3(): Promise<string | undefined> {
   const s3location = launchpadTokenBucketKey();
   const keyExists = await s3ObjectExists(s3location);
 
-  let token = null;
+  let token;
   if (keyExists) {
     const s3object = await getS3Object(s3location.Bucket, s3location.Key);
-    const launchpadToken = JSON.parse(s3object.Body.toString());
-
-    // check if token is still valid
-    if (Date.now() / 1000 < launchpadToken.session_maxtimeout + launchpadToken.session_starttime) {
-      token = launchpadToken.sm_token;
+    if (s3object && s3object.Body) {
+      const launchpadToken = <TokenObject>JSON.parse(s3object.Body.toString());
+
+      // check if token is still valid
+      if (
+        Date.now() / 1000 < launchpadToken.session_maxtimeout + launchpadToken.session_starttime
```

Comment:
This would be easier to read if the `launchpadToken.session_maxtimeout + launchpadToken.session_starttime` part were calculated into a temporary variable, and converted to ms, before the comparison.